verifyml/model_tests/FEAT/tests/test_SubgroupDisparity.py

```diff
@@ -0,0 +1,178 @@
+
+    # nothing should be saved
+    assert len(test_obj1.plots) == 0
+    assert len(test_obj2.plots) == 0
+
+
+def test_run_ratio():
+    """Test that calling .run() updates the test object's .result and .passed attributes."""
+    # init test objects
+    test_obj1 = SubgroupDisparity(
+        attr="gender", metric="fpr", method="ratio", threshold=1.4,
+    )
+
+    test_obj2 = SubgroupDisparity(
+        attr="gender", metric="fnr", method="ratio", threshold=1.5,
+    )
+
+    test_obj3 = SubgroupDisparity(
+        attr="gender", metric="mse", method="ratio", threshold=1.5,
+    )
+
+    test_obj4 = SubgroupDisparity(
+        attr="gender", metric="mae", method="ratio", threshold=1.5,
+    )
+
+    # run tests
+    test_obj1.run(test_data)
+    test_obj2.run(test_data)
+    test_obj3.run(test_data)
+    test_obj4.run(test_data)
+
+    assert test_obj1.result.iloc[0].gender_fpr_max_ratio == 1.5
+    assert test_obj1.passed == False
+
+    assert test_obj2.result.iloc[0].gender_fnr_max_ratio == 1.333  # rounded to 3 d.p
+    assert test_obj2.passed == True
+
+    assert test_obj3.result.iloc[0].gender_mse_max_ratio == 1.5
+    assert test_obj3.passed == True
+
+    assert test_obj4.result.iloc[0].gender_mae_max_ratio == 1.5
+    assert test_obj4.passed == True
+
+
+def test_run_difference():
+    """Test that calling .run() updates the test object's .result and .passed attributes."""
+    # init test objects
+    test_obj1 = SubgroupDisparity(
+        attr="gender", metric="fpr", method="diff", threshold=0.2,
+    )
+
+    test_obj2 = SubgroupDisparity(
+        attr="gender", metric="fnr", method="diff", threshold=0.1,
+    )
+
+    test_obj3 = SubgroupDisparity(
+        attr="gender", metric="mse", method="diff", threshold=0.1,
+    )
+
+    test_obj4 = SubgroupDisparity(
+        attr="gender", metric="mae", method="diff", threshold=0.1,
+    )
+
+    # run tests
+    test_obj1.run(test_data)
+    test_obj2.run(test_data)
+    test_obj3.run(test_data)
+    test_obj4.run(test_data)
+
+    assert test_obj1.result.iloc[0].gender_fpr_max_diff == 0.167  # rounded to 3 d.p
+    assert test_obj1.passed == True
+
+    assert test_obj2.result.iloc[0].gender_fnr_max_diff == 0.167  # rounded to 3 d.p
+    assert test_obj2.passed == False
+
+    assert test_obj3.result.iloc[0].gender_mse_max_diff == 0.2
+    assert test_obj3.passed == False
+
+    assert test_obj4.result.iloc[0].gender_mae_max_diff == 0.2
+    assert test_obj4.passed == False
+
+
+def test_run_chi2():
+    """Test that calling .run() updates the test object's .result and .passed attributes."""
+    # init test objects
+    test_obj1 = SubgroupDisparity(
+        attr="gender", metric="fpr", method="chi2", threshold=0.05,
+    )
+
+    test_obj2 = SubgroupDisparity(
+        attr="gender", metric="fnr", method="chi2", threshold=0.6,
+    )
+
+    # run tests
+    test_obj1.run(test_data)
+    test_obj2.run(test_data)
+
+    assert test_obj1.result.iloc[0].p_value == 1
```

Comment:
Can we change the example, to have a p-value < 1. Otherwise, these two test are not that meaningful. With two different thresholds, one should pass and the other fail.

Reply:
Both still have a p-value of 1